popsycle/synthetic.py

```diff
@@ -3512,3 +3527,390 @@ def generate_slurm_scripts(slurm_config_filename, popsycle_config_filename,
         print(stderr)
 
         print('Submitted job {0} to {1}'.format(script_filename, resource))
+
+def angdist(ra1, dec1, ra2, dec2):
+    '''
+    Takes two spherical coordinates and determines the angluar distance between
+    them.
+    Input:
+    ra1,dec1 [degrees] angular coordinates of the first point
+    ra2,dec2 [degrees] angular coordinates of the second point
+    Output:
+    distance [degrees] angular distance between the two points
+    '''
+    from numpy import pi, sin, cos, arccos
+        d2r = pi / 180.0
+        cosdist = (cos(np.absolute(ra1 - ra2) * d2r) * cos(np.absolute(dec1 * d2r)) * cos(np.absolute(dec2 * d2r)) +
+                   sin(np.absolute(dec1 * d2r)) * sin(np.absolute(dec2 * d2r)))
+        distance = arccos(cosdist) / d2r
+        return distance
+
+
+def add_pbh(hdf5_file, output_root2, area, fdm, pbh_mass, r_max, c, r_vir, overwrite = False, seed = None):
+    """
+    Given some hdf5 file from perform_pop_syn output, creates PBH positions, velocities, etc,
+    and saves them in a new HDF5 file with the PBHs added.
+
+    Parameters
+    ----------
+    hdf5_file : str or hdf5 file
+        str : name of the hdf5 file from the output of perform_pop_syn
+
+    output_root2 : str
+        The thing you want the output files to be named
+        Examples:
+           'myout'
+           '/some/path/to/myout'
+           '../back/to/some/path/myout'
+
+    area : float 
+        Area of the sky that will be generated, in square degrees.
+
+    fdm : float
+        Fraction of dark matter. 
+        The fraction of dark matter that you want to consist of PBHs.
+        Defaults to 1.
+
+    pbh_mass : int
+        The single mass that all PBHs will have.
+        Defaults to 40 Msun (from LIGO detections thought to be primordial)
+
+    r_max : float
+        The maximum radius from the galactic center that you want to find PBHs at.
+        Defaults to 8.3 kpc (Where Earth is located)
+
+    c : float
+        Concentration index.
+        Defaults to 10 (concentration index of the Milky Way, multiple papers show it to be anywhere from 10-12)
+
+    r_vir : float
+        The virial radius.
+        Defaults to 200 kpc (The virial radius of the Milky Way)
```

Comment:
If these are stated as having default values, then they need to have defaults assigned to them in the arguments of `add_pbh`

Reply:
These defaults are listed in the pbh_config.yaml file. I was under the impression that if you didn't specify one of the values, it would pull it from the config file. Is the not the case? Do I still need to assign them in add_pbh? And if I do still need to assign them in add_pbh, what is the point of pbh_config.yaml? (As I thought that was its function, I am confused now).